### PR TITLE
Adds end-to-end Wireguard client test.

### PR DESCRIPTION
**Pull request by cpu**
This PR updates the Vagrant test environment to provision the streisand-client machine as a Wireguard client. The connection to the streisand-server is tested, along with the changes to the client DNS.

To do this I abstracted out the wireguard install tasks into a separate yml file in the wireguard role so the test role can include it without duplication. The wg0-client.conf template also had to be updated to use the hack from https://github.com/jlund/streisand/pull/696 so that a client can reliably wg-quick up the interface with openresolv. Since the hack sucks and a better fix is coming I gated it on the client test var.

### CODE REVIEW

**Comment by cpu** [comment]
Example output snipped from a clean `vagrant up`:
```
TASK [test-client : Determine the running kernel release] **********************
changed: [streisand-client]

TASK [test-client : Add the WireGuard PPA] *************************************
ok: [streisand-client]

TASK [test-client : Install the WireGuard packages] ****************************
ok: [streisand-client] => (item=[u'linux-headers-4.4.0-78-generic', u'linux-headers-generic', u'wireguard-dkms', u'wireguard-tools'])

TASK [test-client : Download the Wireguard wg0-client config file] *************
ok: [streisand-client]

TASK [test-client : Register the original /etc/resolv.conf contents] ***********
ok: [streisand-client]

TASK [test-client : Bring up the Wireguard interface] **************************
changed: [streisand-client]

TASK [test-client : Register the modified /etc/resolv.conf contents] ***********
ok: [streisand-client]

TASK [test-client : Assert that the DNS was updated for the DNSMasq Wireguard IP] ***
ok: [streisand-client] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [test-client : Check https://github.com through the Wireguard route] ******
ok: [streisand-client]

TASK [test-client : Bring down the Wireguard interface] ************************
changed: [streisand-client]

TASK [test-client : Re-Register the /etc/resolv.conf contents] *****************
ok: [streisand-client]

TASK [test-client : Assert that the DNS was restored to pre-Wireguard state] ***
ok: [streisand-client] => {
    "changed": false, 
    "msg": "All assertions passed"
}
```

File: playbooks/roles/test-client/tasks/main.yml [hunk 1]
```diff
@@ -93,3 +93,7 @@
 - name: "Run test scripts"
   command: "{{ item.cmd }}"
   with_items: "{{ test_scripts }}"
+
+# Run the Wireguard tests
+- include: wireguard.yml
```
Comment (jlund) [inline playbooks/roles/test-client/tasks/main.yml]: This syntax will apply the tag to every task in the included file which could eliminate some repetition:

```
- include: wireguard.yml
  tags:
    - wireguard
```

File: playbooks/roles/test-client/tasks/main.yml [hunk 2]
```diff
@@ -93,3 +93,7 @@
 - name: "Run test scripts"
   command: "{{ item.cmd }}"
   with_items: "{{ test_scripts }}"
+
+# Run the Wireguard tests
```
Comment (jlund) [inline playbooks/roles/test-client/tasks/main.yml]: Wireguard -> WireGuard

File: playbooks/roles/test-client/vars/main.yml [hunk 3]
```diff
@@ -18,6 +18,9 @@ test_scripts:
   - { template: 'streisand-ssh-forward-test.sh.j2',         cmd: 'streisand-ssh-forward-test' }
   - { template: 'streisand-shadowsocks-forward-test.sh.j2', cmd: 'streisand-shadowsocks-forward-test' }
 
+# An external URL used to test connectivity through Shadowsocks/Wireguard/etc
```
Comment (jlund) [inline playbooks/roles/test-client/vars/main.yml]: -> "Shadowsocks/WireGuard/etc."

File: playbooks/roles/test-client/vars/main.yml [hunk 4]
```diff
@@ -46,3 +49,9 @@ gateway_shadowsocks_client: "{{ gateway_test_url }}/mirror/shadowsocks/shadowsoc
 # Path to the Shadowsocks2 client source on the Streisand client
 shadowsocks_client: "/root/shadowsocks2-linux-x64"
 shadowsocks_client_zip: "{{ shadowsocks_client }}.gz"
+
+# The gateway URL for clients to download the Wireguard config file
```
Comment (jlund) [inline playbooks/roles/test-client/vars/main.yml]: WireGuard

File: playbooks/roles/test-client/vars/main.yml [hunk 5]
```diff
@@ -46,3 +49,9 @@ gateway_shadowsocks_client: "{{ gateway_test_url }}/mirror/shadowsocks/shadowsoc
...
+gateway_wireguard_config: "{{ gateway_test_url }}/wireguard/wg0-client.conf"
+
+# Path to the wireguard config file on the Streisand client
```
Comment (jlund) [inline playbooks/roles/test-client/vars/main.yml]: WireGuard

File: playbooks/roles/test-client/tasks/main.yml (hunk 1 above)
Comment (cpu) [inline playbooks/roles/test-client/tasks/main.yml]: Oh nice! Thanks

File: playbooks/roles/test-client/tasks/main.yml (hunk 2 above)
Comment (cpu) [inline playbooks/roles/test-client/tasks/main.yml]: Amazing I made an effort to catch these and still left 3+ :blush:

**Comment by cpu** [comment]
@jlund Ready for another :mag: when you have a chance.

**Pull request merged**